modules/services/repositories/src/main/java/au/com/shiftyjelly/pocketcasts/repositories/playback/UpNextQueue.kt

```diff
@@ -113,19 +114,35 @@ interface UpNextQueue {
                                 .observePodcastByUuidFlow(state.podcast.uuid)
                                 .distinctUntilChanged { t1, t2 -> t1.isUsingEffects == t2.isUsingEffects },
                         ) { episode, podcast ->
-                            State.Loaded(episode, podcast, state.queue)
+                            val loadedState = State.Loaded(episode, podcast, state.queue)
+                            updateCurrentEpisodeStateIfNeeded(episode, loadedState)
+                            loadedState
                         }
                         .catch { emit(State.Empty) }
                 } else {
                     episodeManager.observeEpisodeByUuid(state.episode.uuid)
-                        .map<BaseEpisode, State> { State.Loaded(it, state.podcast, state.queue) }
+                        .map<BaseEpisode, State> {
+                            val loadedState = State.Loaded(it, state.podcast, state.queue)
+                            updateCurrentEpisodeStateIfNeeded(it, loadedState)
+                            loadedState
+                        }
                         .catch { emit(State.Empty) }
                 }
             } else {
                 flowOf(state)
             }
         }
     }
+
+    fun updateCurrentEpisodeStateIfNeeded(episodeFromDb: BaseEpisode, state: State) {
+        currentEpisode?.let { currentEpisode ->
+            if (episodeFromDb.uuid == currentEpisode.uuid &&
+                episodeFromDb.deselectedChapters.sorted() != currentEpisode.deselectedChapters.sorted()
```

Comment:
Is this check important? I'm genuinely asking, not suggesting. It feels like it shouldn't matter if we update the list with another list if the contain the same elements but in different order.

Reply:
Ah! I was comparing strings in an earlier attempt when sorting was required. With lists, it isn't. Removed in https://github.com/Automattic/pocket-casts-android/pull/1916/commits/615b5fb409089608706ededf170642b299863664